ts/store/reducers/entities/readTransactions.ts

```diff
@@ -0,0 +1,43 @@
+import * as pot from "italia-ts-commons/lib/pot";
+import { values } from "lodash";
+import { getType } from "typesafe-actions";
+import { Action } from "../../actions/types";
+import { readTransaction } from "../../actions/wallet/transactions";
+import { GlobalState } from "../types";
+
+// Search transaction number in Set of read transactions
+export const isReadTransaction = (state: GlobalState, txid: number) =>
+  state.entities.transactionsRead[txid] !== undefined;
+
+// filter only unread transactions to account for the residual number
+export const getUnreadTransactions = (state: GlobalState) =>
+  pot.map(
+    state.wallet.transactions.transactions,
+    txs =>
+      values(txs).filter(
+        _ => _ !== undefined && !isReadTransaction(state, _.id)
+      ) // TODO: convert to ID array to lighten the weight
+  );
+
+export type ReadTransactionsState = Readonly<{
+  [key: string]: number;
+}>;
+
+const INITIAL_STATE: ReadTransactionsState = {};
+
+export const transactionsReadReducer = (
+  state: ReadTransactionsState = INITIAL_STATE,
+  action: Action
+): ReadTransactionsState => {
+  switch (action.type) {
+    case getType(readTransaction): {
+      const transactionsRead = action.payload.id.toString();
```

Comment:
Why "transactionsRead"? Isn't this the transactionId?

Reply:
I chose "transactionsRead" because I added the transaction id to an indexed typed object in order to gather all seen transactions (by Id).
So in the end, this object contains all transactions read.

Reply:
However, I'll change the name to a more comprehensible meaning.